pkg/kubectl/cmd/annotation.go

```diff
@@ -0,0 +1,288 @@
+	return cmd
+}
+
+// Complete adapts from the command line args and factory to the data required.
+func (o *AnnotateOptions) Complete(f *cmdutil.Factory, args []string, out io.Writer) (err error) {
+	o.namespace, _, err = f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	// retrieves resource and annotation args from args
+	for _, s := range args {
+		isAnnotation := strings.Contains(s, "=") || strings.HasSuffix(s, "-")
+		switch {
+		case isAnnotation:
+			o.annotationArgs = append(o.annotationArgs, s)
+		case !isAnnotation:
+			o.resources = append(o.resources, s)
+		}
+	}
+
+	o.newAnnotations, o.removeAnnotations = parseAnnotations(o.annotationArgs)
+
+	mapper, typer := f.Object()
+	o.builder = resource.NewBuilder(mapper, typer, f.ClientMapperForCommand()).
+		ContinueOnError().
+		NamespaceParam(o.namespace).DefaultNamespace().
+		ResourceTypeOrNameArgs(o.all, o.resources...).
+		Flatten().
+		Latest()
+
+	return nil
+}
+
+// Validate checks to the AnnotateOptions to see if there is sufficient information run the command.
+func (o AnnotateOptions) Validate(args []string) error {
+	// checks args to verify that all resources are specified before annotations
+	metAnnotationArgs := false
+	for _, s := range args {
+		isAnnotation := strings.Contains(s, "=") || strings.HasSuffix(s, "-")
+		switch {
+		case !metAnnotationArgs && isAnnotation:
+			metAnnotationArgs = true
+		case metAnnotationArgs && !isAnnotation:
+			return fmt.Errorf("all resources must be specified before annotation changes: %s", s)
+		}
+	}
+	if len(o.resources) < 1 {
+		return fmt.Errorf("one or more resources must be specified as <resource> <name> or <resource>/<name>")
+	}
+	if len(o.annotationArgs) < 1 {
+		return fmt.Errorf("at least one annotation update is required")
+	}
+
+	if err := validateAnnotations(o.annotationArgs, o.removeAnnotations, o.newAnnotations); err != nil {
+		return err
+	}
+
+	// only apply resource version locking on a single resource
+	if len(o.resources) > 1 && len(o.resourceVersion) > 0 {
+		return fmt.Errorf("--resource-version may only be used with a single resource")
+	}
+
+	return nil
+}
+
+// RunAnnotate does the work
+func (o AnnotateOptions) RunAnnotate() error {
+	single := false
+	r := o.builder.Do().IntoSingular(&single)
+	if err := r.Err(); err != nil {
+		return err
+	}
+	return r.Visit(func(info *resource.Info) error {
+		_, err := cmdutil.UpdateObject(info, func(obj runtime.Object) error {
+			err := updateAnnotations(obj, o.overwrite, o.resourceVersion, o.newAnnotations, o.removeAnnotations)
+			if err != nil {
+				return err
+			}
+			return nil
+		})
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+}
+
+// parseAnnotations retrieves new and remove annotations from annotation args
+func parseAnnotations(annotationArgs []string) (newAnnotations map[string]string, removeAnnotations []string) {
```

Comment:
This could return the error that "validate" would produce during the initial parse: https://github.com/GoogleCloudPlatform/kubernetes/pull/11779/files#diff-051990852b771d2bbf3fcd3209a24063R208